### PR TITLE
Issue #13328: Kill mutation for UnusedLocalVariableCheck 10

### PR DESCRIPTION
**Pull request by Kevin222004**
Issue #13328: Kill mutation for UnusedLocalVariableCheck 10

------

# Check :- 
https://checkstyle.org/checks/coding/unusedlocalvariable.html#UnusedLocalVariable

--------

# Mutation
https://github.com/checkstyle/checkstyle/blob/554a87ea9b5ae27f9519e8dbca15a116bcc108a4/config/pitest-suppressions/pitest-coding-2-suppressions.xml#L84-L95

-----------

# Explaination

The reason to make typeast null is as similar as https://github.com/checkstyle/checkstyle/pull/13145#issue-1739850025
that we don't throw violation on instance or class variable

----------

# Regression :- 

Diff Regression config: https://gist.githubusercontent.com/Kevin222004/0dbdfb487f8e9a050ed0e6356f1a35b4/raw/31f7927f400e11e4285e86fd086b373095227848/unusedLocal.xml
Diff Regression projects: https://gist.githubusercontent.com/Kevin222004/9600f179b602d4c971bdb0a050099005/raw/360a95ed7bb60d7a0956e531199d484c4d6f6617/test-projects.properties
Report label: Final-Regression-2

### CODE REVIEW

**Comment by Kevin222004** [comment]
Github, generate report

File: src/main/java/com/puppycrawl/tools/checkstyle/checks/coding/UnusedLocalVariableCheck.java [hunk 1]
```diff
@@ -1020,7 +1020,7 @@ public Deque<VariableDesc> getUpdatedCopyOfVarStack(DetailAST literalNewAst) {
             final Deque<VariableDesc> instAndClassVarDeque = new ArrayDeque<>();
             instanceAndClassVarStack.forEach(instVar -> {
                 final VariableDesc variableDesc = new VariableDesc(instVar.getName(),
-                        instVar.getTypeAst(), updatedScope);
+                        null, updatedScope);
```
Comment (romani) [inline src/main/java/com/puppycrawl/tools/checkstyle/checks/coding/UnusedLocalVariableCheck.java:1023]: Please make two params ctor instead

**Review by romani — CHANGES_REQUESTED** [review]
Items

File: src/main/java/com/puppycrawl/tools/checkstyle/checks/coding/UnusedLocalVariableCheck.java (hunk 1 above)
Comment (Kevin222004) [reply src/main/java/com/puppycrawl/tools/checkstyle/checks/coding/UnusedLocalVariableCheck.java]: Done

**Comment by Kevin222004** [comment]
Github, generate report

**Comment by Kevin222004** [comment]
@romani @rdiachenko @Vyom-Yadav Please merge this pr first before https://github.com/checkstyle/checkstyle/pull/13151 and https://github.com/checkstyle/checkstyle/pull/13152

**Review by romani — APPROVED** [review]
Ok to merge

File: src/main/java/com/puppycrawl/tools/checkstyle/checks/coding/UnusedLocalVariableCheck.java [hunk 2]
```diff
@@ -875,6 +875,18 @@ private VariableDesc(String name) {
...
+         *
+         * @param name name of the variable
+         * @param scope ast of type {@link TokenTypes#SLIST} or
+         *              {@link TokenTypes#LITERAL_FOR} or {@link TokenTypes#OBJBLOCK}
+         *              which is enclosing the variable
+         */
+        private VariableDesc(String name, DetailAST scope) {
+            this(name, null, scope);
+        }
+
```
Comment (Vyom-Yadav) [inline src/main/java/com/puppycrawl/tools/checkstyle/checks/coding/UnusedLocalVariableCheck.java:889]: Ok for killing the mutation. IMO the check design should be slightly modified not to ask for `type` if it is an instance variable.

**Review by Vyom-Yadav — APPROVED** [review]
LGTM!

**Review by rdiachenko — APPROVED** [review]
lgtm

**Pull request merged**